core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1291,11 +1291,11 @@ abstract class RDD[T: ClassTag](
     } else {
       val buf = new ArrayBuffer[T]
       val totalParts = this.partitions.length
-      var partsScanned = 0
+      var partsScanned = 0L
```

Comment:
I think there is a legit problem here. Imagine `totalParts` is close to `Int.MaxValue`, and imagine `partsScanned` is close to `totalParts`. Adding `p.size` to it below could cause it to roll over. I think this change is needed.